### ISSUE TITLE
Enhancement: `set -e` is ignored in a subshell with specific surrounding context

### ISSUE THREAD

**Issue by cuihaoleo** [koalaman/shellcheck#1484]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
(
    set -e
    false
    echo AAA
) && echo BBB
```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

See: https://unix.stackexchange.com/questions/65532/why-does-set-e-not-work-inside-subshells-with-parenthesis-followed-by-an-or

> The -e setting shall be ignored when executing the compound list following the while, until, if, or elif reserved word, a pipeline beginning with the ! reserved word, or any command of an AND-OR list other than the last.

It is a really counter-intuitive behavior. Shellcheck may warn user about ignored `set -e`.
